### PR TITLE
Cancel animation frame requests on component unmount to prevent memory. leaks

### PR DESCRIPTION
**Pull request by Arooba-git**
Hello 👋 
As part of our project, we are using Facebook's new Memlab tool to detect memory leaks in SPA applications and their libraries. While running the tool and analyzing the code of foxglove's studio, we saw that your project does a very good job of ensuring that all async operations are cancelled when components destroy. However, as per Memlab execution results, we found some uncancelled animation frame requests, causing the memory to leak (screenshots below).

[before]
before Screen Shot 2023-02-14 at 1 30 36 AM

Hence we added the fix by cancelling the requests on component unload, and you can see the count of leaks reducing noticeably:
 <br />

after anim Screen Shot 2023-02-14 at 1 35 47 AM

We executed the unit test suite after the fixes in this PR and it successfully passed 120 of 120 total (in watch + non-watch mode); we also ensured all the changes passed the lint.

You can analyze this and other potential leak sources, if you like, by running [Memlab](https://facebook.github.io/memlab/) with a scenario file covering maximum count of use cases.

Following is a sample of the scenario file we used (it needs to be a .js file but attaching here in .txt form):
[foxglove-scenario-memlab.txt](https://github.com/foxglove/studio/files/10727459/foxglove-scenario-memlab.txt)

Note that some other reported leaks (in Memlab) originated from React's internal objects, and hence were ignored.

### CODE REVIEW

File: packages/studio-base/src/components/TimeBasedChart/index.tsx [hunk 1]
```diff
@@ -196,13 +197,15 @@ export default function TimeBasedChart(props: Props): JSX.Element {
     if (current) {
       // allow the chart offscreen canvas to render to screen before calling done
       requestAnimationFrame(current);
+      requestID.current = requestAnimationFrame(current);
```
Comment (jtbandes) [inline packages/studio-base/src/components/TimeBasedChart/index.tsx:200]: This seems wrong, it's adding a 2nd call to rAF.

Comment (Arooba-git) [reply packages/studio-base/src/components/TimeBasedChart/index.tsx:200]: Hmmm, can you elaborate on this a bit? 🤔

Comment (jtbandes) [reply packages/studio-base/src/components/TimeBasedChart/index.tsx:200]:
```
      requestAnimationFrame(current);
      requestID.current = requestAnimationFrame(current);

```

There are now 2 calls to `requestAnimationFrame` when previously there was only 1.

Comment (Arooba-git) [reply packages/studio-base/src/components/TimeBasedChart/index.tsx:200]: Oh right, this is a mistake, let me fix it up!

Comment (Arooba-git) [reply packages/studio-base/src/components/TimeBasedChart/index.tsx]: fixed :)

**Review by jtbandes — APPROVED** [review]

**Comment by jtbandes** [comment]
@defunctzombie please merge

**Pull request merged**